src/stores/reservation.ts

```diff
@@ -0,0 +1,98 @@
+import { defineStore } from "pinia";
+import { getSeanceById } from "@/api/services/Seances";
+import { getMovieById } from "@/api/services/Movies";
+import { getHall } from "@/api/services/Halls.js";
+
+export const useReservationStore = defineStore({
+  id: "reservation",
+  state: (): Store => ({
+    seance: null,
+    seanceIsLoading: false,
+    seanceError: null,
+    movie: null,
+    movieIsLoading: true,
+    movieError: null,
+    hall: null,
+    hallIsLoading: false,
+    hallError: null,
+  }),
+  getters: {
+    getReservationErrorMessage(state) {
+      return (
+        state.seanceError?.message ||
+        "We are sorry, but the seance cannot be displayed."
+      );
+    },
+  },
+  actions: {
+    async getSeance(seanceId: String) {
+      this.seance = null;
+      this.seanceIsLoading = true;
+      this.movie = null;
+      this.movieIsLoading = true;
+      this.hall = null;
+      this.hallIsLoading = true;
+      try {
+        const respData = await getSeanceById(seanceId);
+        this.seance = respData.data;
+      } catch (error) {
+        this.seanceError = error;
+      } finally {
+        this.seanceIsLoading = false;
+      }
+      try {
+        const respData = await getMovieById(this.seance?.movie);
+        this.movie = respData.data;
+      } catch (error) {
+        this.movieError = error;
+      } finally {
+        this.movieIsLoading = false;
+      }
+      try {
+        const respData = await getHall(this.seance?.hall);
+        this.hall = respData.data;
+      } catch (error) {
+        this.hallError = error;
+      } finally {
+        this.hallIsLoading = false;
+      }
+    },
+  },
+});
+
+interface Store {
+  seance: SeanceInfo | null;
+  seanceIsLoading: Boolean;
+  seanceError: null;
+  movie: MovieInfo | null;
+  movieIsLoading: Boolean;
+  movieError: null;
+  hall: HallInfo | null;
+  hallIsLoading: Boolean;
+  hallError: null;
+}
+
+interface SeanceInfo {
+  available_seats: [];
+  datetime: String;
+  hall: Number;
+  id: Number;
+  movie: Number;
+  taken_seats: [];
+}
+
+interface MovieInfo {
+  description: String;
+  genre: {};
+  id: Number;
+  length: Number;
+  poster_url: String;
+  release_date: String;
+  title: String;
+}
+
+interface HallInfo {
+  id: Number;
+  name: String;
+  seats: Number;
+}
```

Comment:
Do you use these types defined here? 🤔

Reply:
Yes I could see the types when I hovered over interface Store and then types in store. Anyway I changed the order so that it is easier to read. Of course I made many mistakes defining types, so I corrected some :).
When it comes to error states I decided to use any type, just in case I didn't know what structure of the error would be. With any type, when I get error that has message key, the message will be returned form getReservationErrorMessage getter. I also tried unknown type - but then the getter would not return the value of message property.
I could also define an interface for error states, since error is an objects (with some nested objects), but I'm not sure if it is worth doing it.